### DISCUSSION TITLE
Hidden videos in download menu.

### DISCUSSION THREAD

**Discussion by MTCMTC000** [aclap-dev/video-downloadhelper#292] [Help and Q&A]
### Your question

I am currently on MacOS 14.2.1, Firefox 123.0.1, DLH 8.2.2.7a1 (beta) and Co-app 2.0.14.  I will be deleting this after I update as much of this as possible.  I do not know if the problem will clear with the updates but it is so strange I wanted to report it now.  

Downloads are disappearing from the drop-down menu but are still downloading.  I have never seen this before.  I have restarted Firefox twice to clear the problem.  It has happened only today but three times.  I will update everything before I try any more downloads.  I cannot provide you any more information because the problem is invisible data.   

### Addon & CoApp version and details

I am currently on MacOS 14.2.1, Firefox 123.0.1, DLH 8.2.2.7a1 (beta) and Co-app 2.0.14

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3296, Firefox 123.0.1 64-bit, licensed VDH 8.2.2.7a1 beta, CoApp 2.0.19.  I don't work for VDH.  I'm just another user.

Comparing our environments, we're on different operating systems, the same Firefox, the same VDH, but different CoApps.  So my first suggestion is to get the latest CoApp 2.0.19 so you're on the latest one.  If that doesn't cure your problem, I might suggest this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0dc9ad25-367b-47c2-bf6e-e4218d44dd3e)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f5c185db-ed81-4e1c-812b-cafef2667499)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/1f8dd070-71e2-4b29-92c1-7cb6d1b4b64f)
![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c842cae2-e2b4-4861-836a-3489f6fc4526)

Also, please post at least one URL on which you are seeing your symptoms.  Let's see if I experience the same symptoms.

**Reply by MTCMTC000** [reply]
I updated the OS and the Co-App.  I don't know why the problem started.  I honestly have a suspicion the ISP may have something to do with the problem.  I noticed a week or two ago, the individual download speeds reported in the app drop-down menu we not accurate.  They had been dead on correct but they changed to a ridiculously high speed as if they were multiplying the real rate.  I then changed to my mobile phone hot spot and the speed report was correct.  Now, while typing this, I am on my mobile phone hot spot again and it is giving me the weird speed reports like the other ISP.  Maybe the computer needs an exorcist.  I'll leave this open for a day to make sure the problem does not return.  Thanks again to GWW for another response to a bug report.

**Reply by GrampaWildWilly** [reply]
Wait.  Are you saying you're not seeing the problem any more?  Was the solution apparently to get on the latest CoApp?

As for the speed thing, there is a problem.  I've pointed it out to Paul & he's aware of it.  When I look at the Windows Resource Monitor while a download is in progress, there are 2 areas of the window that report things.  (Actually 3 areas, but I'm interested in only 2.)  At the bottom, there is a list of what appear to be multiple threads of ffmpeg.  Ffmpeg seems to be rather sophisticated about running multiple instances of itself in parallel.  The VDH status menu appears to be reporting one of the speed numbers from one of the threads.  The top part of the Resource Monitor shows the cumulative total bandwidth usage of all the threads.  That number would be the correct one, but at the moment, that's not the one VDH is reporting.  At least on Windows.  So my observation is that VDH is consistently, & by rather a lot, under-reporting the true download speed.  It's interesting that you say that VDH is over-reporting the true download speed.  This must have something to do with the way the operating system provides the speed information, Mac vs Windows.  But when you connect via your phone mobile hot spot, is that slower than your normal ethernet connection?  I'm assuming in your normal computer connection, you've just got a co-ax or RS232 connection into some form of Internet box, which then connects to your computer, most likely via a CAT-5 or CAT-6 cable.  But your alternative connection is WiFi or Bluetooth to your phone.  I'm guessing  that is slower, which probably influences the multi-threading in ffmpeg. Meaning there's less of it, so VDH's faulty measuring algorithm is getting it right because you're presenting it a simpler environment, a boundary condition, a base case.  Do you have anything like the Resource Monitor on Mac?  Here's what mine looks like at the moment.  Nothing going on just now.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/34b3bd74-8f55-4d1e-b95a-28a654dfba92)

Do you have a tool like this on Mac?  You should watch that while your downloads are in progress & compare what it looks like when you are using your primary connection vs your connection via the phone hot spot.

**Reply by MTCMTC000** [reply]
The problem is not fixed.  I was just talking part of my lunch time to download some videos for later tonight and realized the number of active downloads displayed with the 3 balls icon was stuck at one.  That one download was actually stuck so I canceled it.  Right now, I am watching the MacOS Activity Monitor data transfer graph maxing out the DSL connection at 480 kB/s. I suspect I unwittingly set up a long list of downloads because I was thinking the system was keeping up.  The icon is dark right now but the app seems to be downloading.  I cannot guess what has become disconnected or why.  It was working correctly when I started an hour ago.  I will leave it sit and operate while I get back to work.  Later

**Reply by GrampaWildWilly** [reply]
I would be tempted to do a Reload extension.  You can see that option in one of the images I posted above.  That would completely kill whatever downloads might be running & empty whatever queue you might have.  Then you could start over from a known, clean starting point.  Of course, any partial downloads this leaves behind would likely be useless.

In the past, I occasionally noticed that a download in progress was not listed on the blue dot status menu.  Instead, it was listed directly on the main VDH menu.  It was the correct status, complete with time remaining & a progress bar, but it was in the wrong place.  But that was something I saw only very rarely, & I have not seen it in a very long time.  Also every once in a while, I would cancel a download & it would continue anyway.  My solution for that was Reload extension.  But that is also something I have not seen in quite a long time.  Paul fixed the cancel functionality some time last year & it's worked properly since then.  An invisible download is something I don't believe I've ever encountered.  This is all quite puzzling.

I do hope you changed your Video qualities settings, as I recommended above.  There is one setting in there that limits the number of variants displayed on the VDH menu.  I have always ramped that up to its maximum value.  At the moment, that maximum is 10, which I think is too low.  But you should set that to 10 to make sure you aren't causing this problem for yourself.

**Reply by MTCMTC000** [reply]
Well, now the problem has changed.  I reloaded the extensions.  I don't know that it changed anything but the problem is different.  The icon is showing the number of active downloads, not the total of active and queued.  When on completes, one of the hidden queued downloads pops up.  I did just noticed the drop-down menu list of video qualities shows "queued" after I select the resolution to download but before it starts downloading.  This is not the hidden downloads problem.  If this is a change to the operation of the app, if would help to have had a notice about the updates.  I will say I liked having the queued videos in the list of active downloads.  Does any of this make sense?

**Reply by GrampaWildWilly** [reply]
It's been a while since I did anything that caused a queue to back up.  But as I recall, the count in the blue dot is supposed to include both in progress + queued downloads.  I believe the queued downloads are supposed to be listed on the blue dot status menu but I can't remember whether the inflight downloads are supposed to be above or below the ones waiting on the queue.  The number in the blue dot gets incremented as soon as you start to schedule a download.  That is, as soon as you open the file selection dialog, the number in the blue dot gets increased by 1, so it doesn't increase again when you finally hit Save & the download actually begins.

Why do you have a queue?  Are you requesting download of more items than the value of your Max concurrent downloads setting?

You might advance your cause here by capturing a series of screen shots of what happens.  I am getting a bit lost in the details.

**Reply by MTCMTC000** [reply]
Everything you described seems to fit with the original way the app functions.  Only in the last 24 hours,after more than a decade of using the app, has this changed.  I do not yet have access to fast enough internet to eliminate the need for a queue.  I can only safely run 4 concurrent downloads because of the poor internet service.  Frankly, if I get on fiber soon I will not need to download much of anything.  

I will work on getting some snaps and descriptions tomorrow.  I have thought about reinstalling the app and co-app but this does not seem to be the kind of bug a reinstall will correct.

Thank you

**Reply by MTCMTC000** [reply]
I am starting to wonder if the problem is related to MacOS.  The last update or two has introduced some minor bugs and I wonder if they slid into app functions.  

The images below show what appears after I have added a fifth to the download list so it is first in the queue.  The second image shows the drop-down menu after one of the original 4 has completed and the "fifth," now forth, has started.  The blue box only shows "4" in both images but should show 5 in the first image.  I forgot to snap the list of active downloads, that used to also include the videos in the queue.   Let me know if you want more information.  

![1  The fifth video, first in the queue](https://github.com/aclap-dev/video-downloadhelper/assets/158776722/e1e73207-fb9d-4ba2-9083-14e0996f9e0b)
![2  After the download has started for this video](https://github.com/aclap-dev/video-downloadhelper/assets/158776722/3910ddef-acf8-4d6a-91f8-23e8d1b6dc71)

**Reply by GrampaWildWilly** [reply]
It might be more helpful to show what it tells you when you click the blue dot with the 4 in it at various stages of progress.

I know about slow connections.  I put up with one for years.  I waited & waited & waited for the local phone company to lay in their much-advertised fiber optic service.  I finally got fed up waiting about 3 years ago & switched to my cable TV provider as my ISP.  It multiplied my speed by a factor of over 100.  From 500,000 bytes per second (if I was lucky) to 62.5 million bytes per second.  And fiber optic service is still not available in this part of the city.  I think about that & always remember to appreciate what I have now.

**Reply by MTCMTC000** [reply]
I still remember the hoops I had to jump through to get a dial up line capable of 56k vs 28k, or something like that.  I need to put in a point-to-point antenna system to connect to the fiber 1800 feet away.  

I am now on my mobile hotspot and I don't want to eat up the data for the month.  It downloads fast enough I can snap images before the download is complete.  I did try.  I will get more images tomorrow.

**Reply by GrampaWildWilly** [reply]
56K.  You were lucky.  The phone lines went into the ground around here I think in the late '70s.  When I tried to get a 56K modem to work at 56K & couldn't, I called the phone company.  They told me that the phone lines were just old enough that they were incapable of carrying data at 56K.  The highest speed I ever saw was 26.6K.  I never even got 28K.  DSL was like a miracle to me.  Of course, the world & the Internet evolved & I outgrew DSL.  It had become quite tortoise-like.  Now I've got what my ISP likes to call 500M, which of course is really 62.5 million bytes per second.  I wonder how many years it will take until I feel like this is tortoise-like.

**Comment by GrampaWildWilly** [comment]
It occurs to me that it might be a good idea to run over something you may or may not know.  And if you know it, maybe somebody else reading this at some point might be learning from this.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8de9dec2-2d24-48b3-bbc3-f3a21a3003ec)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/81147132-6b50-4257-8329-21796b088362)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8a14f40b-840c-4e37-813a-beaa70144e5a)
![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/621b5837-d992-41ce-b0e4-3ce5b0649e6d)

VDH has always behaved like this, at least as far back as VDH 6.x.

**Reply by MTCMTC000** [reply]
I concur with the images above.  However, this is not the way it is functioning now.  I have tried to illustrate this with the images below. 

![1  Showing I have 4 downloads started](https://github.com/aclap-dev/video-downloadhelper/assets/158776722/656ea150-f608-4a87-bd02-75e5a554c25a)
![2  Showing the varient options](https://github.com/aclap-dev/video-downloadhelper/assets/158776722/5f5b74b8-d431-4987-8e48-733ea6ec3a65)
![3  Showing the queued status in the varient list, not the progress list as it should](https://github.com/aclap-dev/video-downloadhelper/assets/158776722/1fb58ce0-1b18-434e-9c53-ecb0b66298c6)

**Reply by GrampaWildWilly** [reply]
OK.  So you've set your queue length at 4.  When you queue a fifth item, it remains on the main menu marked as queued.  What happens when one of the current 4 completes?  Doesn't the fifth download disappear from the main menu & go to the status menu, like the 4 you have now?  And doesn't the one that completes come back to the main menu?  You would see it on the main menu only if you go back to the page from which you launched it.  Otherwise, you could see it by clicking the green dot with the 50 in it.  Hover over each of those dots to get a tooltip to hover up explaining what each of those dots means.  Each one is clickable.

As an aside, I think it is good policy, not a requirement of VDH, just a good practice to get in the habit of doing, to keep open any window from which you launch a VDH download.  That way, you can check its status along the way.  You can also easily redo a download should it fail.  In your case, you can find that download after it completes without too much scrounging around.  No need to click dots.  Just go back to the home page of that download & look at the VDH menu for that page again.  Also, I rarely accept the file name VDH proposes for a download.  I almost always change the file name to something I prefer.

**Reply by GrampaWildWilly** [reply]
Further to my good policy idea, if you close the web page from which you launch a download, then VDH will not have a menu on which to report the completion of a download.  Now, we can argue that VDH needs some sort of persistent activity log.  That's actually a request that has come up a number of times over the years.  I've been among those militating for that.  But the way it works today, you have to keep the home page of the video open in a window or tab.  Then after that download completes, the VDH menu on that home page will show the completed download with the play icon, like I showed in my images upthread.

**Reply by GrampaWildWilly** [reply]
Plus, it appears I need to revise something I said upthread.  It seems the queued downloads don't go on the status menu after all.  In addition, the queued items aren't included in the number on the blue dot.  I wish I had a clearer recollection of how it used to work.  A couple of years ago I was on a web site where I was purposely using the queue restrictions because of some rules imposed by the web site.  But (a) I'm no longer a member of that site (it was a pay site), & (b) that was on whatever VDH was current at the time, probably one of the earlier versions of VDH 7.x.  It would not surprise me that the queue management has changed with the advent of ffmpeg being our download engine in VDH 8.x, or I should say more accurately in CoApp 2.x.  There is another currently open discussion here in which a fellow user is wrestling with VDH not processing the queue in the order in which he is queuing items.  He expects it to be strictly FIFO, which I think is completely reasonable.  But apparently VDH is picking things off the queue for downloading in random order.  You haven't expressed any preference for the order of events in your case, but you might actually be observing this random sequencing.  You might not care so you haven't noticed.  Just an observation . . .

**Reply by GrampaWildWilly** [reply]
That would be #139.

**Reply by MTCMTC000** [reply]
I have been dealing with service outages for about 4 days.  I have been able to get some downloads to succeed.  Your suggestion of keeping the video page open has helped but it does use up band width.  The YouTube viewer will download a cache of part of the video even if it is not playing, but I suspect you know this.  

I can say I have changed the concurrent download quantity and this can lead to some more strange results.  I have had the blue box number be only 2 when concurrent download quantity is set to 5 and there is a queue of more than 5 videos.  I suspect the missing 3 are somehow hidden but actually downloading.  I cannot guess what is causing this strange activity nor why there are not more reports of this.  

I hope to add more information in the next few days.

**Reply by MTCMTC000** [reply]
GWW: What happens when one of the current 4 completes? Doesn't the fifth download disappear from the main menu & go to the status menu, like the 4 you have now?
Me: Yes, but, it appears that when I close the video tab before the video in the tab advances to the status menu it will not appear in the status menu.  I am just now seeing this connection.  I am not 100% certain this is correct.
When the download complete the item does come back to the main menu.  
GWW: Plus, it appears I need to revise something I said upthread. It seems the queued downloads don't go on the status menu after all. In addition, the queued items aren't included in the number on the blue dot. I wish I had a clearer recollection of how it used to work. 
Me: Your previous description is exactly the way VDH worked for me until a week and a half ago.  I am not happy with the new method, specifically that information is no longer visible.  I agree with the desire to have 
GWW: ...VDH not processing the queue in the order in which he is queuing items.
Me: This can be an issue for me but is not critical.  

On a possibly related note, I am having some issues with incomplete downloads.  Most examples seem to be with clipped "shorts."  I cannot find an existing topic that describes this..  The problem is repeatable as far as I can tell.  Should I start a new topic or wait until this one is sorted out?

**Reply by MTCMTC000** [reply]
Does that fact that the app is back to working like normal mean the problem is fixed?  I did update Firefox so maybe the problem had nothing to do with VDH.

**Reply by MTCMTC000** [reply]
I just updated to Firefox ver 124.0.  Hopefully it does change anything.

**Reply by MTCMTC000** [reply]
All appears well now.  I am closing this topic.  THANK YOU ALL!
